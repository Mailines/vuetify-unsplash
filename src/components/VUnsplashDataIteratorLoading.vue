<template>
  <v-card
    class="elevation-0 overflow-y-auto"
    :height="imageHolderHeight + 'px'"
  >
    <v-layout
      class="d-flex justify-center align-center"
      :fill-height="imageHolderHeight - 10 + 'px'"
    >
      <v-progress-circular
        :size="100"
        :width="7"
        :color="accentColor"
        indeterminate
        >{{ loadingLabel }}</v-progress-circular
      >
    </v-layout>
  </v-card>
</template>
<script>
export default {
  props: {
    imageHolderHeight: {
      type: Number,
      required: true
    },
    accentColor: {
      type: String,
      default: "#1976d2",
      required: true
    },
    loadingLabel: {
      type: String,
      default: "Loading...",
      required: true
    }
  }
};
</script>
