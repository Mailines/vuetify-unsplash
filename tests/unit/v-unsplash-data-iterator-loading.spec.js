import VUnsplashDataIteratorLoading from "../../src/components/VUnsplashDataIteratorLoading";
import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";

describe("VUnsplashDataIteratorLoading.vue", () => {
  let localVue;
  let wrapper;
  let vuetify;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    wrapper = mount(VUnsplashDataIteratorLoading, {
      localVue,
      vuetify,
      propsData: {
        loadingLabel: "Loading...",
        imageHolderHeight: 700,
        accentColor: "#1976d2"
      }
    });
  });
  it("renders and match snapshot", async () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
