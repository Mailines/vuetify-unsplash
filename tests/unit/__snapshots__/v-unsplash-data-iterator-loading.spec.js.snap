// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUnsplashDataIteratorLoading.vue renders and match snapshot 1`] = `
<div class="elevation-0 overflow-y-auto v-card v-sheet theme--light" style="height: 700px;">
  <div class="layout d-flex justify-center align-center fill-height">
    <div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-circular v-progress-circular--indeterminate" style="height: 100px; width: 100px; color: rgb(25, 118, 210); caret-color: #1976d2;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="21.505376344086024 21.505376344086024 43.01075268817205 43.01075268817205" style="transform: rotate(0deg);">
        <circle fill="transparent" cx="43.01075268817205" cy="43.01075268817205" r="20" stroke-width="3.0107526881720434" stroke-dasharray="125.664" stroke-dashoffset="125.66370614359172px" class="v-progress-circular__overlay"></circle>
      </svg>
      <div class="v-progress-circular__info">Loading...</div>
    </div>
  </div>
</div>
`;
